workflow/controller/dag.go

```diff
@@ -60,6 +60,17 @@ func (d *dagContext) getTaskFromNode(node *wfv1.NodeStatus) *wfv1.DAGTask {
 		if node.Name == d.taskNodeName(task.Name) {
 			return &task
 		}
+		// It could be that this node resulted from a task generated by an expansion; check if that's the case.
+		expandedTasks, err := expandTask(task)
+		if err != nil {
+			panic("could not expand task " + task.Name + " while looking for task corresponding to node " + node.Name + ": " + err.Error())
```

Comment:
This is a code smell, but I argue it's okay for two reasons:

1. Panics are recovered in the operator, so they fail gracefully: https://github.com/argoproj/argo/blob/c6ef1ff19e1c3f74b4ef146be37e74bd0b748cd7/workflow/controller/operator.go#L159-L168

2. This panic _should_ be unreachable, by the time this function is called `expandTask` will have already been called on all possible tasks. If it resulted in an error, it would have been caught earlier in execution.

Reply:
Code was refactored, this comment is no longer relevant.